File: Back/.Net/Modules/Users/BetFriends.Users.Application/Features/SignIn/SignInHandler.cs
using BetFriends.Shared.Application.Abstractions.Messaging;
using BetFriends.Users.Application.Abstractions;

namespace BetFriends.Users.Application.Features.SignIn;

public sealed class SignInHandler(IAuthenticationGateway authenticationGateway,
                                   IHashPassword hashPassword) : IQueryHandler<SignInQuery, Authentication>
{
    private readonly IAuthenticationGateway authenticationGateway = authenticationGateway;
    private readonly IHashPassword hashPassword = hashPassword;

    public Task<Authentication> Handle(SignInQuery request, CancellationToken cancellationToken)
    {
        var password = hashPassword.Hash(request.Password);
        return authenticationGateway.AuthenticateAsync(new AuthenticationRequest(request.Email, password));
    }
}


public record SignInQuery(string Email, string Password) : IQuery<Authentication>;

public record Authentication(string AccessToken, string RefreshToken);

File: Back/NodeJS/modules/users/src/application/features/sign-in/SignInHandler.ts
import { IRequestHandler } from "../../../../../shared/application/Request/IRequestHandler";
import { IAuthenticationGateway } from "../../abstractions/IAuthenticationGateway";
import { IHashPassword } from "../../abstractions/IHashPassword";
import { SignInRequest } from "./SignInQuery"

export class SignInHandler implements IRequestHandler<SignInRequest, Authenticate>  {
    constructor(private readonly authenticationGateway: IAuthenticationGateway,
                private readonly hashPassword: IHashPassword
    ){}
    Handle(request: SignInRequest): Promise<Authenticate> {
        const password = this.hashPassword.Hash(request.Password)
        return this.authenticationGateway.Authenticate(request.Email, password)
    }
    GetRequestType(): string {
        return SignInRequest.name;
    }
}

export interface Authenticate {
    AccessToken: string;
    RefreshToken: string;
}